packages/belos/tpetra/src/solvers/Belos_Tpetra_GmresSstep.hpp

```diff
@@ -163,6 +165,11 @@ class GmresSstep : public Gmres<SC, MV, OP>  {
       stepSize = params.get<int> ("Step Size");
     }
 
+    bool computeRitzValuesOnFly = this->input_.computeRitzValuesOnFly;
+    if (params.isParameter ("Compute Ritz Values on Fly")) {
```

Comment:
You can just do this:
```c++
computeRitzValuesOnFly = params.get ("Compute Ritz Values on Fly", input_.computeRitzValuesOnFly);
```
More importantly -- is there a reason why users wouldn't want to do this?  If this is `true` by default, when would users ever want it to be `false`?  I would rather expose fewer parameters, unless there is a good reason to expose them.

Reply:
Per our e-mail discussion, it sounds like we don't yet fully understand when and whether setting this option is a good idea.  Thus, it seems reasonable to leave this as an option.  The only question is whether the default value should be `true`, since that is riskier (what if the monomial candidate basis is numerically not a basis?).